### ISSUE TITLE
softmax in GraphormerAttentionHead

### ISSUE THREAD

**Issue by JingweiQu** [leffff/graphormer-pyg#14]
`softmax = torch.softmax(a, dim=-1)`

The softmax function is not exact here since we should compute the attention in each graph. However, such direct computation causes the attention between nodes from different graphs in a batch.

Maybe combining `batch_mask` with -10^6 replacing 0 is a solution.

**Comment by leffff** [comment]
Thanks for the issue!
Yeah, that sounds like a good solution!

**Comment by leffff** [comment]
fixed in https://github.com/leffff/graphormer-pyg/commit/f56d6487f05ffc105a0f977b303ce596338a23fc

You're welcome to test!!!

**Comment by Sorrow-Knight** [comment]
Hi，
in the forward method of  class GraphormerAttentionHead(nn.Module) , I think there are two changes that should be made to ensure that the sum of each row vector after applying softmax function equals 1:

1) Change batch_mask_neg_inf[ptr[i]:ptr[i + 1], ptr[i]:ptr[i + 1]] = 1 to batch_mask_neg_inf[ptr[i]:ptr[i + 1], ptr[i]:ptr[i + 1]] = 0.

2) Change a = (a + b + c) * batch_mask_neg_inf to a = (a + b + c) + batch_mask_neg_inf.

Would you agree with these proposed changes?
